develop-docs/api-server/application-domains/database-migrations/index.mdx

```diff
@@ -145,144 +145,202 @@ This is complicated due to our deploy process. When we deploy, we run migrations
+- Make another PR that:
+    - Checks if the column is either not nullable, or doesn't have a db_default set. If either of these is true, then make it nullable via `null=True`.
+    - If the column is a foreign key, remove the database level foreign key constraint it by setting `db_constraint=False`.
+    - Remove the column and in the generated migration use `SafeRemoveField(..., deletion_action=DeletionAction.MOVE_TO_PENDING)` to replace `RemoveField(...)`. This only marks the state for the column as removed.
+    - Combine these migrations together to save making multiple deploys
+- Deploy your migration changes. It's important that all previous pull requests are in production before we remove the actual column from the table.
+- Make a pull request that create a new migration that has the same `SafeRemoveField` operation as before, but set `deletion_action=DeletionAction.DELETE` instead. This deletes the actual column from the table in Postgres.
+- Deploy the drop column migration.
+
+Here's an example of removing the `project` column from this model. It is both a foreign key and not null:
 
-Here's an example of removing columns that were already nullable. First we remove the columns from the model, and then modify the migration to only update the state and make no database operations.
+```python
+@region_silo_model
+class TestModel(Model):
+    __relocation_scope__ = RelocationScope.Excluded
+
+    project = FlexibleForeignKey("sentry.Project")
+    name = models.TextField()
+
+    class Meta:
+        app_label = "uptime"
+        db_table = "uptime_testmodel"
+```
+
+First we remove the constraint and make the column nullable:
+
+```python
+# Model change
+...
+project = FlexibleForeignKey("sentry.Project", db_constraint=False, null=True)
+...
+
+# Migration operations
+operations = [
+    migrations.AlterField(
+        model_name="testmodel",
+        name="project",
+        field=sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+            db_constraint=False,
+            null=True,
+            on_delete=django.db.models.deletion.CASCADE,
+            to="sentry.project",
+        ),
+    ),
+]
+```
+
+Once we've done this, we can now remove the column from the model and generate the migration to remove it. The generated migration looks like this:
+
+```python
+operations = [
+    migrations.RemoveField(model_name="testmodel", name="project"),
+]
+```
+
+Deleting this way is unsafe, so we want to replace this with `SafeDeleteModel` instead. So this becomes
+
+```python
+operations = [
+    SafeRemoveField(model_name="testmodel", name="project", deletion_action=DeletionAction.MOVE_TO_PENDING),
+]
+```
+
+Using `SafeRemoveField` allows us to remove all the state related to the column, but defer deleting it until a later migration. So now, we can combine the migration to remove the constraints and delete the column state together like so
```

Comment:
It's a little confusing here that the next codeblock we're saying "That second migration you just made, throw that one away, we can combine them into the first"

We do need to describe both, since in the case where there's no alter needed for removing null or FKs, we just want the safe remove, but I feel like we should tweak this a bit to make it more clear that that's whats happening here

Reply:
Maybe it'd help to just post the two migrations necessary first, then write the steps to get to that state after?

Reply:
I think that would make sense

Reply:
@evanpurkhiser I updated the docs for deleting a model. Let me know if it's an improvement, and then I'll update the column docs too